applications/FluidDynamicsApplication/tests/cpp_tests/test_alternative_qs_vms_dem_coupled_element.cpp

```diff
@@ -0,0 +1,138 @@
+    for (ModelPart::NodeIterator it_node=model_part.NodesBegin(); it_node<model_part.NodesEnd(); ++it_node){
+        it_node->AddDof(VELOCITY_X,REACTION_X);
+        it_node->AddDof(VELOCITY_Y,REACTION_Y);
+        it_node->AddDof(VELOCITY_Z,REACTION_Z);
+        it_node->AddDof(PRESSURE,REACTION_WATER_PRESSURE);
+        double& r_fluid_fraction = it_node->FastGetSolutionStepValue(FLUID_FRACTION);
+        r_fluid_fraction = 1.0;
+        Matrix& r_permeability = it_node->FastGetSolutionStepValue(PERMEABILITY);
+        r_permeability = ZeroMatrix(Dim, Dim);
+        for (unsigned int d = 0; d < Dim; ++d){
+            r_permeability(d,d) = 1.0e+30;
+        }
+    }
+
+    std::vector<ModelPart::IndexType> element_nodes {1, 2, 4, 3};
+    model_part.CreateNewElement("AlternativeQSVMSDEMCoupled2D4N", 1, element_nodes, p_properties);
+
+    // Loop starts at 1 because you need one less clone than time steps (JC)
+    for (unsigned int i = 1; i < buffer_size; i++) {
+        model_part.CloneTimeStep(i * delta_time);
+    }
+
+    // Define the nodal values
+    Matrix reference_velocity(4,2);
+    reference_velocity(0,0) = 0.0; reference_velocity(0,1) = 0.1;
+    reference_velocity(1,0) = 0.1; reference_velocity(1,1) = 0.2;
+    reference_velocity(2,0) = 0.2; reference_velocity(2,1) = 0.3;
+    reference_velocity(3,0) = 0.3; reference_velocity(3,1) = 0.4;
+
+
+    Geometry<Node<3>>& r_geometry = model_part.ElementsBegin()->GetGeometry();
+
+
+    for(unsigned int i=0; i<4; i++){
+        r_geometry[i].FastGetSolutionStepValue(PRESSURE)    = 0.0;
+        r_geometry[i].FastGetSolutionStepValue(PRESSURE, 1) = 0.0;
+        for(unsigned int k=0; k<2; k++){
+            r_geometry[i].FastGetSolutionStepValue(VELOCITY)[k]    = reference_velocity(i,k);
+            r_geometry[i].FastGetSolutionStepValue(VELOCITY, 1)[k] = 0.9*reference_velocity(i,k);
+            r_geometry[i].FastGetSolutionStepValue(MESH_VELOCITY)[k]    = 0.0;
+            r_geometry[i].FastGetSolutionStepValue(MESH_VELOCITY, 1)[k] = 0.0;
+        }
+    }
+
+    // RHS and LHS
+    Vector RHS = ZeroVector(12);
+    Matrix LHS = ZeroMatrix(12,12);
+
+    std::vector<double> output = {-2.665425819,-1.87894198,-0.02477280423,-10.27651236,-5.037560437,-0.05013494554,-19.87147169,-19.57971097,-0.06709466598,-14.8532568,-21.17045328,-0.05799758425}; // QSVMSDEMCoupled2D4N
+
+    for (ModelPart::ElementIterator i = model_part.ElementsBegin(); i != model_part.ElementsEnd(); i++) {
+        const auto& r_process_info = model_part.GetProcessInfo();
+        i->Initialize(r_process_info); // Initialize constitutive law
+
+        const auto& rElem = *i;
+        rElem.Check(r_process_info);
+        i->CalculateLocalVelocityContribution(LHS, RHS, r_process_info);
+
+        //std::cout << i->Info() << std::setprecision(10) << std::endl;
+        //KRATOS_WATCH(RHS);
+
+        for (unsigned int j = 0; j < output.size(); j++) {
+            KRATOS_CHECK_NEAR(RHS[j], output[j], 1e-6);
+        }
```

Comment:
Use `KRATOS_CHECK_VECTOR_NEAR`. I'd also check something from the LHS, the first row should be enough. Something like
```c++
    KRATOS_CHECK_VECTOR_NEAR(row(LHS,0), output_LHS, 1e-6);
```

Reply:
Ok fine, it is strange because in my local computer the tests run ok, maybe these new ones are being ignored. Do you know how to check it and to ensure they are running?

Reply:
are you compiling in `FullDebug`

Reply:
I am running the tests in `FullDebug` and I am getting this:

`Ran 74 tests in 94.725s

OK (skipped=25)`

I think the new tests are ignored in my local machine, but I do not know why.

Reply:
Are you running the tests with the `test_FluidDynamicsApplication.py` script or with a custom one? If you're using your custom script typical error is to forget importing the corresponding application (it happened to me many times :sweat: )

Reply:
Now I created a custom script and I am able to run the new tests created. The errors should be fixed soon.